podman/domain/config.py

```diff
@@ -62,17 +67,45 @@ class PodmanConfig:
     def __init__(self, path: Optional[str] = None):
         """Read Podman configuration from users XDG_CONFIG_HOME."""
 
+        self.is_default = False
         if path is None:
             home = Path(xdg.BaseDirectory.xdg_config_home)
-            self.path = home / "containers" / "containers.conf"
+            self.path = home / "containers" / "podman-connections.json"
+            old_toml_file = home / "containers" / "containers.conf"
+            self.is_default = True
+        # this elif is only for testing purposes
+        elif "@@is_test@@" in path:
+            test_path = path.replace("@@is_test@@", '')
+            self.path = Path(test_path) / "podman-connections.json"
+            old_toml_file = Path(test_path) / "containers.conf"
```

Comment:
this sounds like an ugly hack but I am not a python programmer so I don't really care if this is the right thing to solve it.

Reply:
I agree, I really dislike it too but I tried everything with the mock Env options but it just wouldn't pick up the env I was setting for `XDG_CONFIG_HOME`.
I can look into it further and open a PR in the future if I can figure it out.